### PR TITLE
BPH HLT DQM Offline

### PR DESCRIPTION
**Pull request by anigamova**
First version of DQMOfflineHLT code for BPH

### CODE REVIEW

File: DQMOffline/Trigger/plugins/BPHMonitor.cc [hunk 1]
```diff
@@ -0,0 +1,342 @@
+#include "DQMOffline/Trigger/plugins/BPHMonitor.h"
+
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+
+#include "DQM/TrackingMonitor/interface/GetLumi.h"
+
+#include "CommonTools/TriggerUtils/interface/GenericTriggerEventFlag.h"
+
+
+double MAX_PHI = 3.2, MAX_ETA = 2.6, MAX_PT = 200, MAX_D0 = 100, MAX_Z0 = 150;
```
Comment (mtosi) [inline DQMOffline/Trigger/plugins/BPHMonitor.cc]: please, make eta, pt, d0 and z0 as input parameters

File: DQMOffline/Trigger/plugins/BPHMonitor.cc [hunk 2]
```diff
@@ -0,0 +1,342 @@
...
+//  constructors and destructor
+// -----------------------------
+
+BPHMonitor::BPHMonitor( const edm::ParameterSet& iConfig ) : 
+  folderName_             ( iConfig.getParameter<std::string>("FolderName") )
+  , muoToken_             ( mayConsume<reco::MuonCollection>       (iConfig.getParameter<edm::InputTag>("muons")     ) )  
+  , bsToken_              ( mayConsume<reco::BeamSpot>             (iConfig.getParameter<edm::InputTag>("beamSpot")))
+  , PVsToken_             ( mayConsume<reco::VertexCollection>     (iConfig.getParameter<edm::InputTag>("offlinePVs")))
+  , met_variable_binning_ ( iConfig.getParameter<edm::ParameterSet>("histoPSet").getParameter<std::vector<double> >("metBinning") )
+  , met_binning_          ( getHistoPSet   (iConfig.getParameter<edm::ParameterSet>("histoPSet").getParameter<edm::ParameterSet>   ("metPSet")    ) )
```
Comment (mtosi) [inline DQMOffline/Trigger/plugins/BPHMonitor.cc]: do you really need `met_variable_binning_` and `met_binning` ?

File: DQMOffline/Trigger/plugins/BPHMonitor.cc [hunk 3]
```diff
@@ -0,0 +1,342 @@
...
+  
+//  edm::Handle<reco::GsfElectronCollection> eleHandle;
+//  iEvent.getByToken( eleToken_, eleHandle );
+//  std::vector<reco::GsfElectron> electrons;
+//  if ( int(eleHandle->size()) < nelectrons_ ) return;
+//  for ( auto const & e : *eleHandle ) {
+//    if ( eleSelection_( e ) ) electrons.push_back(e);
+//  }
+//  if ( int(electrons.size()) < nelectrons_ ) return;
+
```
Comment (mtosi) [inline DQMOffline/Trigger/plugins/BPHMonitor.cc]: please, clean this part

File: DQMOffline/Trigger/plugins/BPHMonitor.cc [hunk 4]
```diff
@@ -0,0 +1,342 @@
...
+  edm::Handle<reco::MuonCollection> muoHandle;
+  iEvent.getByToken( muoToken_, muoHandle );
+  if ( int(muoHandle->size()) < nmuons_ ) return;
+  std::vector<reco::Muon> muons;
+  for ( auto const & m : *muoHandle ) {
+    if ( muoSelection_( m ) ) muons.push_back(m);
+  }
+  if ( int(muons.size()) < nmuons_ ) return;
+  for (int i=0;i<muons.size()+1;i++) {
+    muPhi_.denominator->Fill(muons[i].phi());
```
Comment (mtosi) [inline DQMOffline/Trigger/plugins/BPHMonitor.cc]: I'm confused, why are you filling the histo for *all* muons ?

File: DQMOffline/Trigger/plugins/BPHMonitor.cc [hunk 5]
```diff
@@ -0,0 +1,342 @@
...
+    muEta_.denominator->Fill(muons[i].eta());
+    muPt_.denominator ->Fill(muons[i].pt());
+    const Track * track = 0;
```
Comment (mtosi) [inline DQMOffline/Trigger/plugins/BPHMonitor.cc]: shouldn't be reco::Track ? (I do not see `using namespace reco`)

File: DQMOffline/Trigger/plugins/BPHMonitor.cc [hunk 6]
```diff
@@ -0,0 +1,342 @@
...
+    if (muons[i].isTrackerMuon()) track = & * muons[i].innerTrack();
+    else if (muon.isStandAloneMuon()) track = & * muons[i].outerTrack();
+    if (track) {
+      double d0 = track->dxy(beamSpot->position());
```
Comment (mtosi) [inline DQMOffline/Trigger/plugins/BPHMonitor.cc]: are you sure you want to compute the impact parameter w.r.t. the beamspot ?
I thought you want w.r.t. the PV, but I might be wrong

File: DQMOffline/Trigger/plugins/BPHMonitor.cc [hunk 7]
```diff
@@ -0,0 +1,342 @@
...
+      double z0 = track->dz(beamSpot->position());
+      mud0_.denominator ->Fill(d0);
+      muz0_.denominator ->Fill(z0);
+    }
+  }
+  // filling histograms (denominator)  
+//  int ls = iEvent.id().luminosityBlock();//TODO LumiBlock, do we need that? 
+//  metVsLS_.denominator -> Fill(ls, met);
```
Comment (mtosi) [inline DQMOffline/Trigger/plugins/BPHMonitor.cc]: if it is not used, please clean

File: DQMOffline/Trigger/plugins/BPHMonitor.cc (hunk 1 above)
Comment (mtosi) [inline DQMOffline/Trigger/plugins/BPHMonitor.cc]: in addition, I would suggest to have histograms w/ variable binning for both the pT and eta
probably the same applies to d0 and dz

File: DQMOffline/Trigger/plugins/BPHMonitor.h [hunk 8]
```diff
@@ -0,0 +1,127 @@
...
+
+  METME JpsiPhi_;
+  METME JpsiPhi_;
+  METME JpsiEta_;
+  METME JpsiEta_;
+  METME JpsiPt_;
+  METME JpsiPt_;
+  METME JpsiM_;
+  METME JpsiM_;
+
```
Comment (mtosi) [inline DQMOffline/Trigger/plugins/BPHMonitor.h]: why do you have twice the same variable ?

File: DQMOffline/Trigger/plugins/BPHMonitor.h [hunk 9]
```diff
@@ -0,0 +1,127 @@
...
+
+  GenericTriggerEventFlag* num_genTriggerEventFlag_;
+  GenericTriggerEventFlag* den_genTriggerEventFlag_;
+
+//  StringCutObjectSelector<reco::MET,true>         metSelection_;
+//  StringCutObjectSelector<reco::PFJet,true   >    jetSelection_;
+// StringCutObjectSelector<reco::GsfElectron,true> eleSelection_;
```
Comment (mtosi) [inline DQMOffline/Trigger/plugins/BPHMonitor.h]: please, clean

File: DQMOffline/Trigger/plugins/BPHMonitor.h [hunk 10]
```diff
@@ -0,0 +1,127 @@
...
+  METME muz0_;
+
+  METME JpsiPhi_;
+  METME JpsiPhi_;
+  METME JpsiEta_;
+  METME JpsiEta_;
+  METME JpsiPt_;
+  METME JpsiPt_;
+  METME JpsiM_;
+  METME JpsiM_;
```
Comment (mtosi) [inline DQMOffline/Trigger/plugins/BPHMonitor.h]: I do not see these histograms in the source code...

File: DQMOffline/Trigger/python/BPHMonitoring_Client_cff.py [hunk 11]
```diff
@@ -0,0 +1,22 @@
...
+    efficiency     = cms.vstring(
+        "effic_muPhi       'mu efficiency vs phi; mu phi [rad]; efficiency' muPhi_numerator       muPhi_denominator",
+        "effic_muEta       'mu efficiency vs eta; mu eta [rad]; efficiency' muEta_numerator       muEta_denominator",
+        "effic_muPt       'mu efficiency vs pt; mu pt [GeV]; efficiency' muPt_numerator       muPt_denominator",
+        "effic_mu_d0       'mu efficiency vs d0; mu d0 [cm]; efficiency' mu_d0_numerator       mu_d0_denominator",
+        "effic_mu_z0       'mu efficiency vs z0; mu z0 [cm]; efficiency' mu_z0_numerator       mu_z0_denominator",
+    ),
+#    efficiencyProfile = cms.untracked.vstring(
+#        "effic_met_vs_LS 'MET efficiency vs LS; LS; PF MET efficiency' metVsLS_numerator metVsLS_denominator"
+#    ),
```
Comment (mtosi) [inline DQMOffline/Trigger/python/BPHMonitoring_Client_cff.py]: if not used, please clean

**Comment by davidlange6** [comment]
Closing PR. CMSSW_9_0_X development has completed (CMSSW_9_2_X is for data taking)

**Pull request closed without merging**